src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1034,26 +1048,29 @@ protected void GenerateFindFirstChar()
 
                     if ((_anchors & RegexFCD.End) != 0)
                     {
-                        Label l1 = DefineLabel();
-                        Ldthisfld(s_runtextposField);
-                        Ldthisfld(s_runtextendField);
-                        Bge(l1);
-                        Ldthis();
-                        Ldthisfld(s_runtextendField);
-                        Stfld(s_runtextposField);
-                        MarkLabel(l1);
+                        if (minRequiredLength <= 0) // if it's > 0, we already output a more stringent check
```

Comment:
When is it less than 0? I can't figure that out.

Reply:
Never.  But it might be 0.  I thought this was clearer as a way of dividing the possibilities into 2 rather than 3, but I can change it.

Reply:
Yeah, if you don't mind, I think it's confusing especially as ComputeMinLength clearly doesn't return < 0